src/state.rs

```diff
@@ -0,0 +1,295 @@
+                    } else {
+                        bail!("Package {acc} not found")
+                    }
+                }
+            })
+    }
+
+    #[inline]
+    pub fn intern_path<A: AsRef<str>>(
+        &mut self,
+        path: &[A],
+        keyword: bool,
+        create_unknown_packges: bool,
+    ) -> Result<SymbolRef> {
+        self.intern_fold(Symbol::root(keyword).into(), path, create_unknown_packges)
+    }
+
+    #[inline]
+    pub fn intern_relative_path<A: AsRef<str>>(
+        &mut self,
+        path: &[A],
+        create_unknown_packges: bool,
+    ) -> Result<SymbolRef> {
+        self.intern_fold(self.current_package.clone(), path, create_unknown_packges)
+    }
+
+    #[inline]
+    pub fn minimal() -> Self {
+        Self::new_with_package(Package::new(SymbolRef::new(Symbol::root_sym())))
+    }
+
+    pub fn init_lurk_state() -> Self {
+        let mut root_package = Package::new(SymbolRef::new(Symbol::root_sym()));
+
+        // bootstrap the keyword package
+        let keyword_package = Package::new(SymbolRef::new(Symbol::root_key()));
+
+        // bootstrap the lurk package
+        let mut lurk_package = Package::new(root_package.intern(LURK_PACKAGE_SYMBOL_NAME.into()));
+        LURK_PACKAGE_SYMBOLS_NAMES.iter().for_each(|symbol_name| {
+            lurk_package.intern(symbol_name.to_string());
+        });
+
+        // bootstrap the meta package
+        let mut meta_package = Package::new(lurk_package.intern(META_PACKAGE_SYMBOL_NAME.into()));
+        META_PACKAGE_SYMBOLS_NAMES.iter().for_each(|symbol_name| {
+            meta_package.intern(symbol_name.to_string());
+        });
+
+        // bootstrap the user package
+        let mut user_package = Package::new(lurk_package.intern(USER_PACKAGE_SYMBOL_NAME.into()));
+        user_package
+            .use_package(&lurk_package)
+            .expect("all symbols in the lurk package are importable");
+
+        // initiate the state with the lurk user package then add the others
+        let mut state = Self::new_with_package(user_package);
+        state.add_package(root_package);
+        state.add_package(keyword_package);
+        state.add_package(lurk_package);
+        state.add_package(meta_package);
+        state
+    }
+}
+
+#[inline]
+pub fn lurk_sym(name: &str) -> Symbol {
+    Symbol::sym(&[LURK_PACKAGE_SYMBOL_NAME, name])
+}
+
+#[inline]
+pub fn meta_package_symbol() -> Symbol {
+    lurk_sym(META_PACKAGE_SYMBOL_NAME)
+}
+
+#[inline]
+pub fn user_sym(name: &str) -> Symbol {
+    Symbol::sym(&[LURK_PACKAGE_SYMBOL_NAME, USER_PACKAGE_SYMBOL_NAME, name])
+}
+
+static INITIAL_LURK_STATE_CELL: OnceCell<State> = OnceCell::new();
+
+pub fn initial_lurk_state() -> &'static State {
+    INITIAL_LURK_STATE_CELL.get_or_init(State::init_lurk_state)
+}
+
+const LURK_PACKAGE_SYMBOL_NAME: &str = "lurk";
+const USER_PACKAGE_SYMBOL_NAME: &str = "user";
+const META_PACKAGE_SYMBOL_NAME: &str = "meta";
+
+const LURK_PACKAGE_SYMBOLS_NAMES: [&str; 36] = [
```

Comment:
In the fullness of time, I would expect to see these Very Special Symbols™ manifest in something like a `lurk.lurk` file processed at compile time. 😄